src/test/java/streams/part2/exercise/Exercise3.java

```diff
@@ -0,0 +1,44 @@
+package streams.part2.exercise;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.StringJoiner;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise3 {
+
+    @Test
+    void createLimitedStringWithOddNumbersSeparatedBySpaces() {
+        int countNumbers = 10;
+
+        String result = IntStream.iterate(1, i -> i + 2)
+                .limit(countNumbers)
+                .mapToObj(String::valueOf)
+                .collect(() -> new StringJoiner(" "), StringJoiner::add, StringJoiner::merge)
+                .toString();
+
+        assertThat(result, is("1 3 5 7 9 11 13 15 17 19"));
+    }
+
+    @Test
+    void extractEvenNumberedCharactersToNewString() {
+        String source = "abcdefghijklm";
+        AtomicInteger index = new AtomicInteger(0);
```

Comment:
That's excess, you can use just
```
String result = IntStream.range(0, source.length())
                         .filter(i -> i % 2 == 0)
                         .mapToObj(source::charAt)
                         .collect(...)
                               ...
```
Because:
1) It's ordered stream 
2) Result of operations above won't depend on parallel/sequential mode.

Reply:
Yes, you are right.
Proposed solve is better because it doesn't depends on type of stream (parallel / sequential). 
While the original one can generate wrong result on the parallel stream.